sturdycouscous/Garbanzo/Checker.py

```diff
@@ -52,59 +71,110 @@ def get_all_tls_contexts(self):
 
 	# Returns a list of all TLS verions supported for the given domain. 
 	def tls_versions_checker(self, domain):
-		# Contexts for each TLS version
-		# tlsv1_3_context = ssl.create_default_context()
-		# tlsv1_3_context.options |= (ssl.PROTOCOL_TLS | ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1 | ssl.OP_NO_TLSv1_2)
-
-		# tlsv1_2_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
-		# tlsv1_1_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_1)
-		# tlsv1_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-
-		# contexts_dict = {
-		# 	tlsv1_context: "tlsv1",
-		# 	tlsv1_1_context: "tlsv1_1",
-		# 	tlsv1_2_context: "tlsv1_2",
-		# 	tlsv1_3_context: "tlsv1_3",
-		# }
-
-		# contexts = [tlsv1_3_context, tlsv1_2_context, tlsv1_1_context, tlsv1_context]
-		
+		# Contexts for each TLS version		
 		contexts_dict, contexts = self.get_all_tls_contexts()
 		versions_supported = []
 
 		for context in contexts:
 			if self.test_tls_version(context, domain):
 				versions_supported.append(contexts_dict[context])
 
-		return {domain: versions_supported}
+		return versions_supported
 
 
-	def test_tls_version(self, context, domain):
+	def test_tls_version(self, context, domain, port=443):
 		success = True
 		try:
 			s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)	
 			sslSocket = context.wrap_socket(s, server_hostname = domain)
-			sslSocket.connect((domain, 443))
-			print(sslSocket.version())
+			sslSocket.connect((domain, port))
+			# print(sslSocket.version())
 		except:
 			success = False
 		finally:
+			sslSocket.close()
 			return success
 	
 
-	# Returns a list of all ciphers supported by... context...? 
-	# Need more clarification on this one.
-	def get_all_ciphers(self, domain):
-
+	# Returns a dictionary mapping tls version -> ciphers supported on the client side.
+	# socket.shared_ciphers returns only the client-side ciphers.
+	# therefore, need to test one by one.
+	def get_all_ciphers(self):
+		domain = "www.google.com"
```

Comment:
We would probably want the domain passed as an argument rather than hardcoded as google